nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/SendTrapSNMP.java

```diff
@@ -0,0 +1,269 @@
+
+    public static final AllowableValue GENERIC_TRAP_TYPE_1 = new AllowableValue("1", "Warm Start",
+            "A warmStart trap signifies that the sending protocol entity is reinitializing itself such that" +
+                    " neither the agent configuration nor the protocol entity implementation is altered.");
+
+    public static final AllowableValue GENERIC_TRAP_TYPE_2 = new AllowableValue("2", "Link Down",
+            "A linkDown trap signifies that the sending protocol entity recognizes a failure in one of " +
+                    "the communication links represented in the agent's configuration.");
+
+    public static final AllowableValue GENERIC_TRAP_TYPE_3 = new AllowableValue("3", "Link Up",
+            "A linkUp trap signifies that the sending protocol entity recognizes that one of the communication " +
+                    "links represented in the agent's configuration has come up.");
+
+    public static final AllowableValue GENERIC_TRAP_TYPE_4 = new AllowableValue("4", "Authentication Failure",
+            "An authenticationFailure trap signifies that the sending protocol entity is the addressee of a " +
+                    "protocol message that is not properly authenticated.  While implementations of the SNMP must be " +
+                    "capable of generating this trap, they must also be capable of suppressing the emission of such traps " +
+                    "via an implementation- specific mechanism.");
+
+    public static final AllowableValue GENERIC_TRAP_TYPE_5 = new AllowableValue("5", "EGP Neighbor Loss",
+            "An egpNeighborLoss trap signifies that an EGP neighbor for whom the sending protocol entity was " +
+                    "an EGP peer has been marked down and the peer relationship no longer obtains.");
+
+    public static final AllowableValue GENERIC_TRAP_TYPE_6 = new AllowableValue("6", "Enterprise Specific",
+            "An enterpriseSpecific trap signifies that a particular enterprise-specific trap has occurred which " +
+                    "can be defined in the Specific Trap Type field.");
+
+    public static final PropertyDescriptor SNMP_MANAGER_HOST = new PropertyDescriptor.Builder()
+            .name("snmp-trap-manager-host")
+            .displayName("SNMP Manager Host")
+            .description("The host where the SNMP Manager sends the trap.")
+            .required(true)
+            .defaultValue("localhost")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor SNMP_MANAGER_PORT = new PropertyDescriptor.Builder()
+            .name("snmp-trap-manager-port")
+            .displayName("SNMP Manager Port")
+            .description("The port where the SNMP Manager listens to the incoming traps.")
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.PORT_VALIDATOR)
```

Comment:
Based on the standard PORT_VALIDATOR, is the default value of `0` invalid?

Reply:
Yes it is, but unless it's changed, it seems that defaults are not validated.